src/js/stores/__tests__/UserSettingsStore-test.js

```diff
@@ -47,4 +47,64 @@ describe("UserSettingsStore", function() {
       );
     });
   });
+
+  describe("setJSONEditorExpandedSetting", function() {
+    beforeEach(function() {
+      thisPrevSet = LocalStorageUtil.set;
+
+      LocalStorageUtil.get = function() {
+        return JSON.stringify({});
+      };
+      LocalStorageUtil.set = jasmine.createSpy();
+    });
+
+    afterEach(function() {
+      LocalStorageUtil.set = thisPrevSet;
+    });
+
+    it("sets the JSONEditor key to true", function() {
+      UserSettingsStore.setJSONEditorExpandedSetting(true);
+      expect(LocalStorageUtil.set).toHaveBeenCalledWith(
```

Comment:
what we're testing is that `UserSettingsStore` calls `LocalStorageUtil`. that does not make me very confident that our code works. 

we could make those tests more meaningful if we only tested against the public interface of `UserSettingsStore`, right?

like "set to `a`, get value, is it `a`? set to `b`, get value, is it `b`?".

Reply:
we discussed that offline. i still think that tests that are tied to implementation details are not a good practice. i might be wrong though.

Reply:
Let's try to discuss this in a bigger group. I feel you and we definitely want to get a better test quality. I'll do a follow up to this after the discussion to  unblock the PR right now, if that is okay for you.